checker.go

```diff
@@ -305,6 +305,51 @@ func (c *isNilChecker) Negate(got interface{}, args []interface{}) error {
 	return errors.New("the value is nil, but should not")
 }
 
+// HasLen is a Checker checking that the provided value has the provided length.
+// For instance:
+//
+//     c.Assert([]int{42, 47}, qt.HasLen, 2)
+//     c.Assert(myMap, qt.HasLen, 42)
+//
+var HasLen Checker = &hasLenChecker{
+	numArgs: 1,
+}
+
+type hasLenChecker struct {
+	numArgs
+}
+
+// Check implements Checker.Check by checking that len(got) == args[0].
+func (c *hasLenChecker) Check(got interface{}, args []interface{}) (err error) {
+	want, ok := args[0].(int)
+	if !ok {
+		return BadCheckf("expected a numeric length to compare the value to, got %T instead", args[0])
+	}
+	v := reflect.ValueOf(got)
+	switch v.Kind() {
+	case reflect.Array, reflect.Chan, reflect.Map, reflect.Slice, reflect.String:
+	default:
+		return BadCheckf("expected a type with a lenght, got %T instead", got)
+	}
+	if length := v.Len(); length != want {
+		return fmt.Errorf("the provided value has not the expected length of %d:\n(value)\n\t%#v\n(-got length +want lenght)\n\t-: %d\n\t+: %d", want, got, length, want)
```

Comment:
s/lenght/length/

I wonder whether it might be good to print the contents of the slice too.

Reply:
Already there.